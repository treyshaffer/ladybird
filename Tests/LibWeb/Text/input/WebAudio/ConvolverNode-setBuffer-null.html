<!DOCTYPE html>
<title>ConvolverNode setBuffer null test</title>
<script src="../include.js"></script>
<script>
test(() => {
    const context = new AudioContext();
    const conv = context.createConvolver();
    
    println("=== Testing setting buffer to null ===");
    
    // First set a valid buffer
    const buffer = context.createBuffer(1, 128, context.sampleRate);
    conv.buffer = buffer;
    println("Set buffer, numberOfChannels: " + conv.buffer.numberOfChannels);
    
    // Now set to null
    try {
        conv.buffer = null;
        println("PASS: Setting buffer to null did not throw");
        println("Buffer is null: " + (conv.buffer === null));
    } catch (e) {
        println("FAIL: Setting buffer to null threw " + e.name);
    }
});
</script>