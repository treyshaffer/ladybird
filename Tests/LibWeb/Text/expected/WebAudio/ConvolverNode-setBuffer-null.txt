=== Testing setting buffer to null ===
Set buffer, numberOfChannels: 1
PASS: Setting buffer to null did not throw
Buffer is null: true